worker/instancemutater/mutater_test.go

```diff
@@ -156,6 +156,9 @@ func (s *mutaterSuite) info(profiles []string, rev int, add bool) *apiinstancemu
 			{ApplicationName: "lxd-profile",
 				Revision: rev,
 			},
+			{ApplicationName: "app-no-profile",
+				Revision: rev,
+			},
```

Comment:
How does this expose the bug? Is it just that with 2 it would fail because the second app would get evaluated and then override the pointer (and then the test fails because it isn't applying it)

Reply:
Yes, this adds another unit to the unit data which does not have an lxd profile in it, causing the pointer override.